client/src/components/shared/node/_utils/useDrag.js

```diff
@@ -14,27 +14,33 @@ const useDrag = (position, index, isEndpointNode) => {
 
   const { height, width } = useWindowDimensions();
 
-  const handleMouseDown = useCallback(({ clientX, clientY }) => {
-    dispatch.data.changeNodeCoordinates({
-      x: (clientX / width) * 100,
-      y: (clientY / height) * 100,
-      index,
-      isEndpointNode,
-    });
-    setState((state) => ({
-      ...state,
-      isDragging: true,
-    }));
-  }, []);
+  const handleMouseDown = useCallback(
+    ({ clientX, clientY }) => {
+      dispatch.data.changeNodeCoordinates({
+        x: (clientX / width) * 100,
+        y: (clientY / height) * 100,
+        index,
+        isEndpointNode,
+      });
+      setState((state) => ({
+        ...state,
+        isDragging: true,
+      }));
+    },
+    [height, width, index, isEndpointNode]
```

Comment:
Adding `height` and `width` arguments to `handleMouseDown` actually just fixed a huge problem I was having haha. Now when a user resizes the screen it doesnt mess up the nodes. 

See my other comment about the significance of the second `useEffect` argument.

Reply:
Good to know at least one thing got fixed :)